builtin/providers/aws/resource_aws_cloudfront_web_distribution.go

```diff
@@ -0,0 +1,587 @@
+			ViewerProtocolPolicy: aws.String(d.Get("viewer_protocol_policy").(string)),
+			MinTTL:               aws.Int64(int64(d.Get("minimum_ttl").(int))),
+			MaxTTL:               aws.Int64(int64(d.Get("maximum_ttl").(int))),
+			DefaultTTL:           aws.Int64(int64(d.Get("default_ttl").(int))),
+			TrustedSigners: &cloudfront.TrustedSigners{
+				Enabled:  aws.Bool(false),
+				Quantity: aws.Int64(0),
+			},
+			SmoothStreaming: aws.Bool(d.Get("smooth_streaming").(bool)),
+			AllowedMethods: &cloudfront.AllowedMethods{
+				Quantity: aws.Int64(int64(len(allowedMethods))),
+				Items:    allowedMethods,
+				CachedMethods: &cloudfront.CachedMethods{
+					Quantity: aws.Int64(int64(len(cachedMethods))),
+					Items:    cachedMethods,
+				},
+			},
+		},
+		CacheBehaviors: &cloudfront.CacheBehaviors{
+			Quantity: aws.Int64(0),
+		},
+		CustomErrorResponses: &cloudfront.CustomErrorResponses{
+			Quantity: aws.Int64(0),
+		},
+	}, nil
+}
+
+// resourceAwsCloudFrontWebDistributionWaitUntilDeployed blocks until the
+// distribution is deployed. It currently takes exactly 15 minutes to deploy
+// but that might change in the future.
+func resourceAwsCloudFrontWebDistributionWaitUntilDeployed(
+	d *schema.ResourceData, meta interface{}) error {
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"InProgress", "Deployed"},
+		Target:     "Deployed",
+		Refresh:    resourceAwsCloudFrontWebDistributionStateRefreshFunc(d, meta),
+		Timeout:    40 * time.Minute,
+		MinTimeout: 15 * time.Second,
+		Delay:      10 * time.Minute,
+	}
+
+	_, err := stateConf.WaitForState()
+	return err
+}
+
+func resourceAwsCloudFrontWebDistributionStateRefreshFunc(
+	d *schema.ResourceData, meta interface{}) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		v, err := resourceAwsCloudFrontWebDistributionDistributionRetrieve(d, meta)
+
+		if err != nil {
+			log.Printf("Error on retrieving CloudFront distribution when waiting: %s", err)
+			return nil, "", err
+		}
+
+		if v == nil {
+			return nil, "", nil
+		}
+
+		return v.Distribution, *v.Distribution.Status, nil
+	}
+}
+
+func resourceAwsCloudFrontWebDistributionDistributionRetrieve(
+	d *schema.ResourceData, meta interface{}) (*cloudfront.GetDistributionOutput, error) {
+	cloudfrontconn := meta.(*AWSClient).cloudfrontconn
+
+	req := &cloudfront.GetDistributionInput{
+		Id: aws.String(d.Id()),
+	}
+
+	res, err := cloudfrontconn.GetDistribution(req)
+	if err != nil {
+		return nil, fmt.Errorf("Error retrieving CloudFront distribution: %s", err)
+	}
+
+	return res, nil
+}
+
+func resourceAwsCloudFrontWebDistributionAwsStringLists(in interface{}) []*string {
+	list := in.([]interface{})
+	out := make([]*string, 0, len(list))
+	for _, r := range list {
+		s := r.(string)
+		out = append(out, aws.String(s))
+	}
+	return out
+}
+
+func resourceAwsCloudFrontWebDistributionHandleMethods(in interface{}) []*string {
+	// Terraform schemas does not currently support arrays as default values
```

Comment:
While I understand the pain here, would it help if we just omit the default value completely from the API request? Both `allowed_methods` & `cached_methods` are set as `Computed` & `Optional` already, so if this is the default value that AWS sets by default if nothing is passed to the API, it may be "cleaner".